apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactsView.kt

```diff
@@ -0,0 +1,448 @@
+    val chat1Type = getContactType(chat1)
+    val chat2Type = getContactType(chat2)
+
+    when {
+        chat1Type.ordinal < chat2Type.ordinal -> -1
+        chat1Type.ordinal > chat2Type.ordinal -> 1
+
+        else -> chat2.chatInfo.chatTs.compareTo(chat1.chatInfo.chatTs)
+    }
+}
+
+@Composable
+private fun ModalData.DeletedContactsView(rh: RemoteHostInfo?, close: () -> Unit) {
+    ModalView(
+        close = close
+    ) {
+        Column(
+            Modifier.fillMaxSize(),
+        ) {
+            Box(contentAlignment = Alignment.Center) {
+                val bottomPadding = DEFAULT_PADDING
+                AppBarTitle(
+                    stringResource(MR.strings.deleted_chats),
+                    hostDevice(rh?.remoteHostId),
+                    bottomPadding = bottomPadding
+                )
+            }
+
+            ContactsLayout(
+                contactActions = {},
+                contactTypes = listOf(ContactType.REMOVED),
+            )
+        }
+    }
+}
+
+@Composable
+private fun ContactActionsSection(contactActions: @Composable () -> Unit, rh: RemoteHostInfo?) {
+    contactActions()
+    Spacer(Modifier.height(DEFAULT_PADDING))
+
+    val archived = remember { contactChats(chatModel.chats, listOf(ContactType.REMOVED)) }
+
+    if (archived.isNotEmpty()) {
+        SectionView(padding = PaddingValues(bottom = DEFAULT_PADDING)) {
+            SectionItemView(
+                click = {
+                    ModalManager.center.showCustomModal { close -> DeletedContactsView(
+                        rh = rh,
+                        close = close)
+                    }
+                }
+            ) {
+                Icon(
+                    painterResource(MR.images.ic_folder_open),
+                    contentDescription = stringResource(MR.strings.deleted_chats),
+                    tint = MaterialTheme.colors.secondary,
+                )
+                TextIconSpaced(extraPadding = true)
+                Text(text = stringResource(MR.strings.deleted_chats), color = MaterialTheme.colors.onBackground)
+            }
+        }
+    }
+}
+
+@Composable
+private fun ContactsLayout(
+    contactActions: @Composable () -> Unit,
+    contactTypes: List<ContactType>,
+    contactListTitle: String? = null) {
+
+    SectionView {
+        ContactsList(
+            contactTypes = contactTypes,
+            contactActions = contactActions,
+            contactListTitle = contactListTitle
+        )
+    }
+}
+
+@Composable
+fun ContactsView(
+    contactActions: @Composable () -> Unit,
+    rh: RemoteHostInfo?
+) {
+    ContactsLayout(
+        contactActions = { ContactActionsSection(contactActions, rh) },
+        contactTypes = listOf(ContactType.CARD, ContactType.RECENT, ContactType.REQUEST),
+        contactListTitle = stringResource(MR.strings.contact_list_header_title).uppercase()
+    )
+}
+
+@Composable
+private fun ContactsSearchBar(
```

Comment:
on desktop after pasting a known contact address it requires 2 clicks on that contact to open it (first click reveals action buttons, only second click opens chat)

Reply:
it seems like focus re-render cancels the click events with the movement of the screen. https://github.com/simplex-chat/simplex-chat/pull/4479/commits/1dbfb7aeea0a993d16b9a11723bc712386621b35 seems to be an hack that works on the issue